### PR TITLE
Force SSE2 floats for i686 targets

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Just GCC and Clang, I don't have a way to test MSVC behaviour.

Ref: #565

### CODE REVIEW

**Comment by ruevs** [comment]
The use of x87 fp does _not_ cause loss of precision. On the countrary!
https://gcc.gnu.org/wiki/x87note
Therefore we should treat the crash as a bug and figure out why it is happening with x87 floating point instead of hiding it with SSE.

**Comment by nabijaczleweli** [comment]
Right

File: CMakeLists.txt [hunk 1]
```diff
@@ -80,11 +80,17 @@ endif()
 if(MINGW)
     set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -static-libgcc")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libgcc -static-libstdc++")
+endif()
```
Comment (whitequark) [inline CMakeLists.txt]: A comment here would help. Something like:
```suggestion

# Ensure that all platforms use 64-bit IEEE floating point operations for consistency; this is
# most important for the testsuite, which compares savefiles directly and depends on consistent
# rounding of intermediate results.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -80,11 +80,17 @@ endif()
...
 
-    if(TRIPLE STREQUAL "i686-w64-mingw32")
-        set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -msse2")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2")
+if((CMAKE_SYSTEM_PROCESSOR STREQUAL "i686" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "X86"))
```
Comment (whitequark) [inline CMakeLists.txt]:
```suggestion
if(CMAKE_SYSTEM_PROCESSOR STREQUAL "i686" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "X86")
```

**Comment by ruevs** [comment]
OK makes sense. The "compares savefiles directly" explains it.

**Comment by nabijaczleweli** [comment]
Review addressed

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**